### ISSUE TITLE
Is there a way to set instance level defaults.headers after instance created? (defaults.headers is globally)

### ISSUE THREAD

**Issue by up9cloud** [axios/axios#1117]
#### Summary

I'm trying to setup clients for different headers by https://github.com/axios/axios#custom-instance-defaults.

But defaults.headers is globally.

Is there a way to set headers by instance level?

```js
const axios = require('axios')

axios.defaults.headers.common['x-custom-token'] = 'i0'

let instance1 = axios.create()
// do something...
instance1.defaults.headers.common['x-custom-token'] = 'i1'

let instance2 = axios.create()
// do something...
instance2.defaults.headers.common['x-custom-token'] = 'i2'

console.log(axios.defaults.headers.common)
//{ Accept: 'application/json, text/plain, */*',
//  'x-custom-token': 'i2' }
console.log(instance1.defaults.headers.common)
//{ Accept: 'application/json, text/plain, */*',
//  'x-custom-token': 'i2' }
console.log(instance2.defaults.headers.common)
//{ Accept: 'application/json, text/plain, */*',
//  'x-custom-token': 'i2' }
```

#### Context

- axios version: 

```sh
$ cat node_modules/axios/package.json |jq .version
"0.16.2"
```

- Environment: 

```sh
$ node -v
v8.1.3
```

**Comment by emilyemorehouse** [comment]
Closing, as this issue was resolved in #1395. 🙌

**Comment by Frondor** [comment]
@emilyemorehouse you forgot to close this actually.
